black.py

```diff
@@ -190,6 +191,25 @@ def supports_feature(target_versions: Set[TargetVersion], feature: Feature) -> b
     return all(feature in VERSION_TO_FEATURES[version] for version in target_versions)
 
 
+def merge_exclude(ctx: click.Context, param: click.Parameter, value: str) -> str:
+    """Injects .gitignore in excluded files"""
+    root = find_project_root(ctx.params.get("src", ()))
+    # nothing to do if no .gitignore
+    gitignore = root / ".gitignore"
+    if not gitignore.is_file():
+        return value
+    # parse gitignore
+    excluded_in_gitignore = []
+    for line in gitignore.read_text().splitlines():
+        line = line.strip().strip("/")
+        if line and not line.startswith("#"):
+            excluded_in_gitignore.append(  # noqa  W605 incorrect
+                fnmatch.translate(line).lstrip("(?s:").rstrip(r")\Z")
+            )
+    # merge gitignore and others exclude
+    return r"/(" + r"|".join(excluded_in_gitignore) + "|" + value.strip()[2:]
```

Comment:
`value` is not necessarily going to be in the format `/(something)/` so assuming that will blow up when somebody just passed `--exclude=.git`.

Reply:
Indeed. 
the gitignore values could be merged with :  a regex value (default_include and pyproject) or a more simple regex '--exclude=.git'.
That's not obvious to that we should guess which pattern it is to merge our gitigore's items in it. There could be many pattern so it's not very predictable.

Solutions ? :
- never merge gitignore with the others and try to match twice in `gen_python_files_in_dir`: one as it is now and one and one for gitignore
- Enforce an exclude "style".  say to people to always have at least "/()/" in their regex.
- guess the regex pattern but I don't like it.

Reply:
I think the first solution will be the most robust one.